src/Compilers/CSharp/Test/Emit3/Semantics/ExtensionTests.cs

```diff
@@ -24753,4 +24752,593 @@ public static void M<U>(U u) { }
+DefinitelyAssignedOnExit: i, this, i2
+ReadInside:
+ReadOutside: i, i2
+WrittenInside: i, i2
+WrittenOutside: this
+""", analysisResults);
+        }
+
+        [Fact]
+        public void RegionAnalysis_06()
+        {
+            var analysisResults = CompileAndAnalyzeDataFlowExpression("""
+static class E
+{
+    extension(int i)
+    {
+        public void M(int i2)
+        {
+            /*<bind>*/i = i2 = 0;/*</bind>*/
+        }
+    }
+}
+""");
+            VerifyDataFlowAnalysis("""
+VariablesDeclared:
+AlwaysAssigned: i, i2
+Captured:
+CapturedInside:
+CapturedOutside:
+DataFlowsIn:
+DataFlowsOut:
+DefinitelyAssignedOnEntry: i, this, i2
+DefinitelyAssignedOnExit: i, this, i2
+ReadInside:
+ReadOutside:
+WrittenInside: i, i2
+WrittenOutside: i, this, i2
+""", analysisResults);
+        }
+
+        [Fact]
+        public void RegionAnalysis_07()
+        {
+            var analysisResults = CompileAndAnalyzeDataFlowExpression("""
+static class E
+{
+    extension(ref int i)
+    {
+        public void M(ref int i2)
+        {
+            /*<bind>*/i = ref i2;/*</bind>*/
+        }
+    }
+}
+""");
+            VerifyDataFlowAnalysis("""
+VariablesDeclared:
+AlwaysAssigned: i
+Captured:
+CapturedInside:
+CapturedOutside:
+DataFlowsIn: i2
+DataFlowsOut: i, i2
+DefinitelyAssignedOnEntry: i, this, i2
+DefinitelyAssignedOnExit: i, this, i2
+ReadInside: i2
+ReadOutside: i, i2
+WrittenInside: i, i2
+WrittenOutside: i, this, i2
+""", analysisResults);
+        }
+
+        [Fact]
+        public void RegionAnalysis_08()
+        {
+            var analysisResults = CompileAndAnalyzeDataFlowExpression("""
+static class E
+{
+    extension(int i)
+    {
+        public void M(int i2)
+        {
+            void local(int i3)
+            {
+                _ = /*<bind>*/i + i2 + i3/*</bind>*/;
+            }
+        }
+    }
+}
+""");
+            VerifyDataFlowAnalysis("""
+VariablesDeclared:
+AlwaysAssigned:
+Captured: i, i2
+CapturedInside: i, i2
+CapturedOutside:
+DataFlowsIn: i3
+DataFlowsOut:
+DefinitelyAssignedOnEntry: i, this, i2, i3
+DefinitelyAssignedOnExit: i, this, i2, i3
+ReadInside: i, i2, i3
+ReadOutside:
+WrittenInside:
+WrittenOutside: i, this, i2, i3
+""", analysisResults);
+        }
+
+        [Fact]
+        public void RegionAnalysis_09()
+        {
+            var analysisResults = CompileAndAnalyzeDataFlowExpression("""
+static class E
+{
+    extension(out int i)
+    {
+        public void M(out int i2)
+        {
+            void local(out int i3)
+            {
+                /*<bind>*/i = i2 = i3 = 0;/*</bind>*/
+            }
+        }
+    }
+}
+""");
+            VerifyDataFlowAnalysis("""
+VariablesDeclared:
+AlwaysAssigned: i, i2, i3
+Captured: i, i2
+CapturedInside: i, i2
+CapturedOutside:
+DataFlowsIn:
+DataFlowsOut: i, i2, i3
+DefinitelyAssignedOnEntry: this
+DefinitelyAssignedOnExit: i, this, i2, i3
+ReadInside:
+ReadOutside: i, i2, i3
+WrittenInside: i, i2, i3
+WrittenOutside: this
```

Comment:
>this [](http://example.com/codeflow?start=16&length=4)

The fact that `this` is mentioned in the result (in any symbol set) feels somewhat unexpected.

Reply:
Consider at least adding a prototype comment